pymongo/mongo_client.py

```diff
@@ -893,7 +893,9 @@ def event_listeners(self):
 
     @property
     def topology_description(self):
-        """(id, topology_type, servers, etc.) of current topology
+        """Returns the current :class:`~pymongo.topology_description.TopologyDescription`.
+        The returned object is immutable, and should be called periodically/as needed
+        to avoid using stale information.
```

Comment:
called --> accessed 
to avoid using stale information --> to avoid seeing stale topology information

Reply:
@ShaneHarvey should we also tell users that it is impossible to guarantee that the `topology_description` won't be changed out from under them while they are accessing this property? Since we are not exposing the topology lock (or even exposing it to users for that matter) it seems impossible to guarantee that accessing this object will provide the latest view of the topology -- conceivably, a monitoring thread could acquire the lock and update the topology while the user holds onto a stale reference. I am not saying we need to provide a way for users to get the 'current'/'in-use' topology description, but it seems like something worth documenting?

Reply:
Good question. Yes the topology can be updated directly after the user calls this method. That will always be the case even if we acquire/release the Topology lock here. I don't think we need to call this out besides what is already written in the docstring.

Reply:
Perhaps "current" is the confusing part. Maybe "Returns the most recent :class:...".

The issue is that the description is stale as soon as it's returned from this method.

Reply:
+1 to changing this to 'most recent'. WE can always expound the precise behavior to any users who end up needing it.